src/cregistry/Makefile.in

```diff
@@ -12,7 +12,7 @@ SQLEXT_OBJS = sqlext.o vercomp.o
 include ../../Mk/macports.autoconf.mk
 
 # required for strdup(3) on Linux and OS X
-CPPFLAGS+=-D_XOPEN_SOURCE=600
+CPPFLAGS+=-D_XOPEN_SOURCE=600 -D_DARWIN_C_SOURCE
```

Comment:
I am not aware of "_DARWIN_C_SOURCE" because this is something specific to Mac. Can you just describe it in a single line? didn't find a convincing answer on google.

Reply:
When specifying any of the feature test flags, such as `-D_XOPEN_SOURCE` or `-D_POSIX_SOURCE`, the macOS header files hide function declarations that are not part of any of these standards. If I remember correctly it as `asprintf(3)` is a non-standard function and thus hidden by `-D_XOPEN_SOURCE`.

I still need to test whether this code now still compiles on Linux – it might need a different define there.